rapid7/metasploit-framework
# Module for authenticated Froxlor RCE (CVE-2023-0315)

PoC: https://shells.systems/froxlor-v2-0-6-remote-command-execution-cve-2023-0315/

This module is based off the above PoC. The vulnerability works like so:

Change the logfile to:`templates/Froxlor/footer.html.twig` using builtin authenticated functionality.
Write a payload to the new logfile by changing the "theme" of the site, again with builtin functionality.
If the theme you send to the server is formatted like so: `{{['COMMAND']|filter('exec')}}` you achieve RCE when you reload the page because the twig file executes your command 🎉 

## Note to Reviewer
Accepting suggestions for my clean up attempt in `on_new_session` for the Unix Command target.

## Verification

List the steps needed to make sure this thing works

1. Start msfconsole
1. Do: `use exploit/linux/http/panos_auth_rce`
1. Set the `RHOST`, `USERNAME`, and `PASSWORD` options
1. Run the module
1. Receive a Meterpreter session as the `root` user.
